test/scripts/dev.test.js

```diff
@@ -455,23 +473,69 @@ function runCommonWithFrontendTests (ref) {
 
-  test('should generate a vscode debug config for actions and web-src', async () => {
-    await ref.scripts.runDev()
-    expect(JSON.parse(vol.readFileSync('/.vscode/launch.json').toString())).toEqual(expect.objectContaining({
-      configurations: [
-        getExpectedActionVSCodeDebugConfig('sample-app-1.0.0/action'),
-        getExpectedActionVSCodeDebugConfig('sample-app-1.0.0/action-zip'),
-        getExpectedUIVSCodeDebugConfig(9080)
-      ]
-    }))
-  })
-
   test('should use https cert/key if passed', async () => {
     const options = { parcel: { https: { cert: 'cert.cert', key: 'key.key' } } }
     const port = 8888
     await ref.scripts.runDev([port], options)
     expect(Bundler.mockServe).toHaveBeenCalledWith(port, options.parcel.https)
   })
+
+  test('should cleanup ui server on SIGINT', async () => {
+    return new Promise(resolve => {
+      testCleanupNoErrors(resolve, ref.scripts, () => {
+        expect(Bundler.mockStop).toHaveBeenCalledTimes(1)
+        expect(mockUIServerInstance.close).toBeCalledTimes(0) // should not be called directly, b/c terminator does
+        expect(mockTerminatorInstance.terminate).toBeCalledTimes(1)
+        expect(httpTerminator.createHttpTerminator).toHaveBeenCalledWith({
+          server: mockUIServerInstance
+        })
+      })
+    })
+  })
+
+  test('should cleanup ui server on error', async () => {
+    await testCleanupOnError(ref.scripts, () => {
+      expect(Bundler.mockStop).toHaveBeenCalledTimes(1)
+      expect(mockUIServerInstance.close).toBeCalledTimes(0) // should not be called directly, b/c terminator does
+      expect(mockTerminatorInstance.terminate).toBeCalledTimes(1)
+      expect(httpTerminator.createHttpTerminator).toHaveBeenCalledWith({
+        server: mockUIServerInstance
+      })
+    })
+  })
+  // eslint-disable-next-line jest/no-test-callback
+  test('should exit with 1 if there is an error in cleanup', async done => {
+    const theError = new Error('theerror')
+    Bundler.mockStop.mockRejectedValue(theError)
+    process.removeAllListeners('SIGINT')
+    process.exit.mockImplementation(() => {
+      expect(mockLogger.error).toHaveBeenCalledWith(theError)
+      expect(process.exit).toHaveBeenCalledWith(1)
+      done()
+    })
+    await ref.scripts.runDev()
+    expect(process.exit).toHaveBeenCalledTimes(0)
+    // send cleanup signal
+    process.emit('SIGINT')
+    // if test times out => means handler is not calling process.exit
+  })
+
+  test('should return another available port for the UI server if used', async () => {
+    mockUIServerAddressInstance.port = 9999
+    const options = { parcel: { https: { cert: 'cert.cert', key: 'key.key' } } }
+    const resultUrl = await ref.scripts.runDev([8888], options)
+    expect(Bundler.mockServe).toHaveBeenCalledWith(8888, options.parcel.https)
+    expect(resultUrl).toBe('https://localhost:9999')
+    expect(mockOnProgress).toHaveBeenCalledWith('Could not use port:8888, using port:9999 instead')
```

Comment:
nit: Not crazy about tests for specific text output.  This means if we decide to change the text, we break the test, I think the 2 previous expectations are sufficient as they test 'function' and not 'form'

Reply:
yup good point 👍